src/api/artifact/abstractor/blob/fetcher.go

```diff
@@ -76,3 +81,19 @@ func (f *fetcher) FetchLayer(repository, digest string) ([]byte, error) {
 	defer reader.Close()
 	return ioutil.ReadAll(reader)
 }
+
+func newRepositoryClient(repository string) (*registry.Repository, error) {
```

Comment:
why?

Reply:
Change is that connect the docker registry directly rather than via core.
When pushing a new image, we need to fetch the manifest to abstract the data

Reply:
So why not make it a shared func?  Or we need to do it after switching to basic auth?